### PR TITLE
KAFKA-4090: Validate SSL connection in client for kafka 3.1.x

### PR DESCRIPTION
**Pull request by azhur**
Cherry-picked @gurinderu's commits made to 3.0.x

### CODE REVIEW

**Review by gaspb — APPROVED** [review]

File: .github/workflows/ci_pr.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
```
Comment (guizmaii) [inline .github/workflows/ci_pr.yml]: `adopt` is deprecated. Can we maybe replace it by `temurin`?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
```
Comment (guizmaii) [inline .github/workflows/release.yml]: `adopt` is deprecated. Can we maybe replace it by `temurin`?

**Review by itsukanov — APPROVED** [review]

**Review by guizmaii — APPROVED** [review]

**Comment by guizmaii** [comment]
@azhur Can we maybe use this fork of the kafka-clients in our fork of zio-kafka?

**Comment by azhur** [comment]
@guizmaii I think that's a good idea, let me create a PR in zio-kafka.

**Pull request merged**